pkg/resiliency/policy.go

```diff
@@ -26,63 +26,68 @@ import (
 
 type (
 	// Operation represents a function to invoke with resiliency policies applied.
-	Operation func(ctx context.Context) error
+	Operation func(ctx context.Context) (any, error)
 
 	// Runner represents a function to invoke `oper` with resiliency policies applied.
-	Runner func(oper Operation) error
+	Runner func(oper Operation) (any, error)
 )
 
-// Policy returns a policy runner that encapsulates the configured
-// resiliency policies in a simple execution wrapper.
+type doneCh struct {
+	res any
+	err error
+}
+
+// Policy returns a policy runner that encapsulates the configured resiliency policies in a simple execution wrapper.
 func Policy(ctx context.Context, log logger.Logger, operationName string, t time.Duration, r *retry.Config, cb *breaker.CircuitBreaker) Runner {
-	return func(oper Operation) error {
+	return func(oper Operation) (any, error) {
```

Comment:
What do you think about wrapping this up into a struct with a type instead of unwrapping `any`? Something (admittedly a little hand-wavey for the sake of a PR comment) like: 

```
type PolicyExecutor[T any] struct {
	wrappedFunc func(context.Context) (T, error)
}

func NewPolicy[T any](wrapped func(context.Context) (T, error)) PolicyExecutor[T] {
	return PolicyExecutor[T]{
		wrappedFunc: wrapped,
	}
}

func (p PolicyExecutor[T]) Execute(ctx context.Context) (T, error) {
	return p.wrappedFunc(ctx)
}

type Result struct {
	StatusCode int
	Message    string
}


// Then we can use it like ...
func main() {
	p := NewPolicy[Result](func(ctx context.Context) (Result, error) {
		return Result{1, "ohai"}, nil
	})
	ctx := context.TODO()
	if res, err := p.Execute(ctx); err != nil {
        fmt.Printf("Oops! %v", err)
	} else {
		fmt.Printf("Result: %v\n", res)
	}

}

```

Reply:
As discussed with John separately, John will work on a refactoring that uses generics in a separate PR.

Reply:
Changed the code so this uses generics now